### PR TITLE
Test Updates

### PR DESCRIPTION
**Pull request by Averynder**
This PR handles the project's 3 different test suites: E2E, client and api tests and aims to update them all

### **Client side**:
- coverage has been increased significantly
- Services and most components are now tested
- Nearly every component is provided with a default usable testing file
- beforeEach() have been replaced to a single synchronous one (async ones were leading to flaky tests)
- massively reduced warning list by including app module within tests

### **E2E side**:
- Some tests for e2e were broken, they were updated based on latest master

### **Server unit test**:
- Unit test were updated based on recent changes done on master
- New Unit tests were added

**Notes**
- any test which doesn't work or is set up improperly will result in a flaky test
- Jasmine / Karma doesn't handle flaky tests well and will give random tests which are failing with the issue
- Any tests which are run individually will fail with ng main, they must be run using 'ng test' and marking the describe with 'f' in front, example: 'fdescribe'

### CODE REVIEW

File: client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts [hunk 1]
```diff
@@ -33,4 +33,69 @@ describe('AssignStockKeepersComponent', () => {
...
+
+  // Test the goBackIventory()
+  // it('Call goBackIventory', () => {
+  //   try {
+  //     component.goBackIventory();
+  //   }
+  //   catch (err) {
+  //   console.error(err);
+  //   }
+  // });
```
Comment (NicholasNagy) [inline client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts:70]: if this isn't needed, let's delete it

File: client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.spec.ts [hunk 2]
```diff
@@ -5,36 +5,114 @@ import { RouterTestingModule } from '@angular/router/testing';
...
+  //       month: ,
+  //       day: ,
+  //       hour: ,
+  //       minute: ,
+  //     }])
+  //   }
+  //   catch (error) {
+  //   console.error(error);
+  //   }
+  // });
```
Comment (NicholasNagy) [inline client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.spec.ts:117]: if this isn't needed, can we delete it?

File: client/src/app/components/employees/employee-list/employee-list.component.spec.ts [hunk 3]
```diff
@@ -30,4 +33,35 @@ describe('ClientGridViewComponent', () => {
...
+  //       id: 2,
+  //       email: 'nick@test.com',
+  //       location: 'Terrebonne',
+  //     }]);
+  //   }
+  //   catch (error) {
+  //     console.error(error);
+  //   }
+  // });
+
```
Comment (NicholasNagy) [inline client/src/app/components/employees/employee-list/employee-list.component.spec.ts:66]: if this isn't needed, can we delete this?

**Review by NicholasNagy — COMMENTED** [review]
So I don't have any comments on the overall design of the tests. We've been continuing to maintain our standards in how we like to do things with fixtures, e2e, etc.

Only concerns are the comments I raised. There's no explanation for some commented out code, so I don't know whether we should remove those or not?

File: client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts (hunk 1 above)
Comment (Averynder) [reply client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts]: @NicholasNagy in response to your comments, for the upcoming changes that will be made the reason these tests are commented out is to show others how to setup their tests / serve as an example

Comment (NicholasNagy) [reply client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts]: okay, cool, can you just add a comment explaining that right above that so it doesn't accidentally get deleted?

File: client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.spec.ts (hunk 2 above)
Comment (daphneaugier) [reply client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.spec.ts:117]: As Avery mentioned above, this is commented out is to show others how to setup their tests / serve as an example.

File: client/src/app/components/employees/employee-list/employee-list.component.spec.ts (hunk 3 above)
Comment (daphneaugier) [reply client/src/app/components/employees/employee-list/employee-list.component.spec.ts:66]: As Avery mentioned above, this is commented out is to show others how to setup their tests / serve as an example

File: client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts (hunk 1 above)
Comment (daphneaugier) [reply client/src/app/components/assign-stock-keepers/assign-stock-keepers.component.spec.ts]: Yes makes sense, for clarifications. Just added the comments in the code.

**Review by NicholasNagy — APPROVED** [review]
Thank you for the associated changes, everything is good, now I will merge.

**Pull request merged**